### PR TITLE
Don't try to look up anonymous users

### PR DESCRIPTION
**Pull request by kgrant-rh**
Fixes an issue where views with `RedHatRolePermission` would fail when unauthenticated users tried to connect.

### CODE REVIEW

**Comment by kgrant-rh** [comment]
@RedHatProductSecurity/corgi-devs FYI

File: corgi/core/authentication.py
Comment (juspence) [inline corgi/core/authentication.py]: What about when a request doesn't have the user attribute defined? Or maybe that's not possible. It would also be great to add unit tests for this case if at all possible.

**Review by juspence — APPROVED** [review]

File: corgi/core/authentication.py
Comment (kgrant-rh) [reply corgi/core/authentication.py]: According to [The DRF docs](https://www.django-rest-framework.org/api-guide/authentication/) :

> If no class authenticates, `request.user` will be set to an instance of `django.contrib.auth.models.AnonymousUser`, and `request.auth` will be set to None.

That being the case, I think request.user will always be an actual user or `AnonymousUser`, for whom `is_authenticated` returns false. I'll make sure there are tests for unauthenticated users though.

**Comment by kgrant-rh** [comment]
@RedHatProductSecurity/corgi-devs Please review.

File: corgi/core/authentication.py
Comment (kgrant-rh) [reply corgi/core/authentication.py]: The tests I've added explicitly use `AnonymousUser`, which is what the DRF authentication will supply to the authentication backend in the case that there's no authentication.

File: tests/test_authentication.py
Comment (juspence) [inline tests/test_authentication.py:75]: I think returning a 403 is correct. You've provided mocked credentials so the user isn't completely unauthenticated (401), but their profile doesn't have any rhat_roles defined so using this endpoint is unauthorized / forbidden (403).

And of course you have the test for unauthenticated users above that returns 401, then you have the test with an authenticated + authorized user below that returns 200. Thank you for adding comprehensive tests!

**Review by juspence — APPROVED** [review]
Still LGTM. Feel free to just merge approved MRs once you resolve all conversations, I don't think a second round of review is required. Although I'm happy to look if you just want a second pair of eyes.

**Pull request merged**
